code-studio/src/js/components/BeeCellEditor.jsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * @overview React component to allow for easy editing and creation of
+ * BeeCells
+ * @see @cdo/apps/maze/beeCell
+ */
+/* global React */
+
+var BeeCell = require('@cdo/apps/maze/beeCell');
+
+var BeeCellEditor = module.exports = React.createClass({
+  propTypes: {
+    cell: React.PropTypes.object.isRequired,
+    row: React.PropTypes.number.isRequired,
+    col: React.PropTypes.number.isRequired,
+    onUpdate: React.PropTypes.func.isRequired,
+  },
+
+  handleChange: function (event) {
+    var values = {};
+    var nodes = this.getDOMNode().querySelectorAll('[name]');
+    for (var i = 0, node; (node = nodes[i]); i++) {
```

Comment:
Anything to raise awareness! (btw, see [Iterating over NodeLists](https://google.github.io/styleguide/javascriptguide.xml?showone=Tips_and_Tricks#Tips_and_Tricks) for my source)

Another approach I've used before and rather like is to declare a helper function in some utils library:

```javascript
var forEachNode = function (nodeList, callback, context) {
  for (var i = 0, node; (node = nodeList[i]); i++) {
    callback.call(context, node, i, nodeList);
  }
};
```